05_pytorch_backend/cnn_acceleration_tensorly_and_pytorch.ipynb

```diff
@@ -0,0 +1,594 @@
+      "    (1): Conv2d (256, 256, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)\n",
+      "    (2): Conv2d (256, 512, kernel_size=(1, 1), stride=(1, 1))\n",
+      "  )\n",
+      "  (22): ReLU(inplace)\n",
+      "  (23): Sequential(\n",
+      "    (0): Conv2d (512, 256, kernel_size=(1, 1), stride=(1, 1), bias=False)\n",
+      "    (1): Conv2d (256, 256, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)\n",
+      "    (2): Conv2d (256, 512, kernel_size=(1, 1), stride=(1, 1))\n",
+      "  )\n",
+      "  (24): ReLU(inplace)\n",
+      "  (25): Sequential(\n",
+      "    (0): Conv2d (512, 256, kernel_size=(1, 1), stride=(1, 1), bias=False)\n",
+      "    (1): Conv2d (256, 256, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)\n",
+      "    (2): Conv2d (256, 512, kernel_size=(1, 1), stride=(1, 1))\n",
+      "  )\n",
+      "  (26): ReLU(inplace)\n",
+      "  (27): MaxPool2d(kernel_size=(2, 2), stride=(2, 2), dilation=(1, 1))\n",
+      "  (28): Sequential(\n",
+      "    (0): Conv2d (512, 256, kernel_size=(1, 1), stride=(1, 1), bias=False)\n",
+      "    (1): Conv2d (256, 256, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)\n",
+      "    (2): Conv2d (256, 512, kernel_size=(1, 1), stride=(1, 1))\n",
+      "  )\n",
+      "  (29): ReLU(inplace)\n",
+      "  (30): Sequential(\n",
+      "    (0): Conv2d (512, 256, kernel_size=(1, 1), stride=(1, 1), bias=False)\n",
+      "    (1): Conv2d (256, 256, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)\n",
+      "    (2): Conv2d (256, 512, kernel_size=(1, 1), stride=(1, 1))\n",
+      "  )\n",
+      "  (31): ReLU(inplace)\n",
+      "  (32): Sequential(\n",
+      "    (0): Conv2d (512, 256, kernel_size=(1, 1), stride=(1, 1), bias=False)\n",
+      "    (1): Conv2d (256, 256, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)\n",
+      "    (2): Conv2d (256, 512, kernel_size=(1, 1), stride=(1, 1))\n",
+      "  )\n",
+      "  (33): ReLU(inplace)\n",
+      "  (34): Sequential(\n",
+      "    (0): Conv2d (512, 256, kernel_size=(1, 1), stride=(1, 1), bias=False)\n",
+      "    (1): Conv2d (256, 256, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)\n",
+      "    (2): Conv2d (256, 512, kernel_size=(1, 1), stride=(1, 1))\n",
+      "  )\n",
+      "  (35): ReLU(inplace)\n",
+      "  (36): MaxPool2d(kernel_size=(2, 2), stride=(2, 2), dilation=(1, 1))\n",
+      ")\n",
+      "('Number of parameters before the decomposition', 20024384L)\n",
+      "('Number of parameters after the decomposition', 7278656L)\n",
+      "('Ratio', 0.3634896334389113)\n"
+     ]
+    }
+   ],
+   "source": [
+    "from functools import reduce # Valid in Python 2.6+, required in Python 3\n",
+    "import operator\n",
+    "\n",
+    "def count_params(model):    \n",
+    "    return  sum(reduce(operator.mul, p.size(), 1) for p in model.parameters())\n",
```

Comment:
since we use numpy already, we could use `np.sum` instead of `sum` and `np.prod` instead of the reduce on mul. This saves us from importing reduce and operator and might be slightly clearer to read :)